src/schema/tablebuilder.js

```diff
@@ -190,6 +190,9 @@ TableBuilder.prototype.foreign = function(column) {
         foreignData.references = pieces ? pieces[0] : tableColumn;
         return {
           on: function(tableName) {
+            if (typeof tableName === 'undefined') {
+              throw new Error("table name is of type 'undefined'. Please pass in a valid table name");
```

Comment:
@ngoctranfire Would you be ok with checking if `typeof tableName !== 'string'` and throwing a `TypeError` instead? It would make more sense considering supplying anything other than a string should be failing anyways.

Error message could be something like `Expected tableName to be a string, got: --typeof tableName--`